src/graph/graph.c

```diff
@@ -57,9 +63,72 @@ void _Graph_ResizeNodes(Graph *g, size_t n) {
     }
 
     g->node_cap = g->block_count * NODEBLOCK_CAP;
-    _Graph_ResizeMatrix(g, g->adjacency_matrix);
 }
 
+/* Relocate src node to dest node, overriding dest. */
+void _Graph_NodeBlockMigrateNode(Graph *g, int src, int dest) {
+    // Get the block in which dest node resides.
+    NodeBlock *destNodeBlock = GRAPH_GET_NODE_BLOCK(g, dest);
+    
+    // Get node position within its block.
+    int destNodeBlockIdx = GRAPH_NODE_POSITION_WITHIN_BLOCK(dest);
+
+    // Get the src node in the graph.
+    Node *srcNode = Graph_GetNode(g, src);
+
+    // Replace dest node with src node.
+    destNodeBlock->nodes[destNodeBlockIdx] = *srcNode;
```

Comment:
I think that the `Node`'s `incoming_edges` and `outgoing_edges` Vectors only get populated for QueryGraph members, but if I am incorrect then they will be broken here.

(Should we consider simplifying that struct?)

Reply:
indeed edges are only used within the context of a query graph, we should create a query node and a query edge, and update the node object.